sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc

```diff
@@ -0,0 +1,976 @@
+of the ordered range resulting from sorting `val` from all work-items in the
+`g` group with respect to the binary comparison function object `comp`
+using the `exec` group helper object.
+For multi-dimensional groups, the order of work-items in the group is
+determined by their linear id.
+
+_Complexity_: Let `N` be the work-group or sub-group size. `O(N*log(N)*log(N))` comparisons.
+
+_Constraints_: Only available if `GroupHelper` was created with a work-group or a sub-group and
+some associated scratch space.
+
+6._Effects_: Equivalent to: `return sorter(g, val)`.
+
+_Constraints_: All functions are available only if `Sorter` is a SYCL Sorter and
+it provides `operator()(Group, T)` overload.
+
+=== Functions with fixed-size arrays
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  template <typename GroupHelper, typename T, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> values, Properties properties = {}); // (1)
+
+  template <typename GroupHelper, typename T, typename U, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> keys, sycl::span<U, ItemsPerWorkItem> values, Properties properties = {}); // (2)
+
+  template <typename GroupHelper, typename T, std::size_t ItemsPerWorkItem, typename Compare, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> values, Compare comp, Properties properties = {}); // (3)
+
+  template <typename GroupHelper, typename T, typename U, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> keys, sycl::span<U, ItemsPerWorkItem> values, Compare comp, Properties properties = {}); // (4)
+
+  template<typename Group, typename T, std::size_t ItemsPerWorkItem, typename Sorter, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(Group g, sycl::span<T, ItemsPerWorkItem> values, Sorter sorter, Properties properties = {}); // (5)
+
+  template<typename Group, typename T, typename U, std::size_t ItemsPerWorkItem, typename Sorter, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(Group g, sycl::span<T, ItemsPerWorkItem> keys, sycl::span<U, ItemsPerWorkItem> values, Sorter sorter, Properties properties = {}); // (6)
+
+}
+----
+
+NOTE: (2), (4), (6) functions below perform sorting including key-value variant.
+Key value sorting is a sorting algorithm where keys are compared,
+but keys and values are reordered both.
+
+1._Constraints_: Only available if `GroupHelper` was created with a work-group or a sub-group and

```

Comment:
The description here has this order:

* Constraints
* Effects
* Complexity

However, the description in the section above has:

* Preconditions
* Effects
* Complexity 
* Constraints

Can you use the same order for each section?